codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/Instantiator.kt

```diff
@@ -252,7 +254,11 @@ class Instantiator(
             writer.rust("$data.to_string()")
         } else {
             val enumSymbol = symbolProvider.toSymbol(shape)
-            writer.rust("#T::from($data)", enumSymbol)
+            if (mode == CodegenMode.Server) {
+                writer.rust("""#T::try_from($data).expect("This is used in tests ONLY")""", enumSymbol)
+            } else {
+                writer.rust("#T::from($data)", enumSymbol)
+            }
```

Comment:
I think we should prefer `when` statements to `if` statements when differentiating between client and server codegen.
- It's very explicit about what cases are being covered
- I don't expect us to ever add a third mode, but writing things this way would force us to update things if we ever did
- It also saves one line of code

```suggestion
            when (mode) {
                 CodegenMode.Server -> writer.rust("""#T::try_from($data).expect("This is used in tests ONLY")""", enumSymbol)
                 CodegenMode.Client -> writer.rust("#T::from($data)", enumSymbol)
            }
```